docker/Dockerfile

```diff
@@ -0,0 +1,63 @@
+# Global ARGs shared by all stages
+ARG DEBIAN_FRONTEND=noninteractive
+ARG GOPATH=/usr/local/go
+
+### first stage - builder ###
+FROM debian:buster-slim as builder
+
+ARG DEBIAN_FRONTEND
+ARG GOPATH
+ENV GOPATH=${GOPATH}
+
+# install debos build dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        ca-certificates \
+        golang-go \
+        gcc \
+        git \
+        libostree-dev \
+        libc6-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+# Build debos
+COPY . $GOPATH/src/github.com/go-debos/debos
+WORKDIR $GOPATH/src/github.com/go-debos/debos/cmd/debos
+RUN go get -d ./... && \
+    go get -d github.com/stretchr/testify && \
+    go install
+
+### second stage - runner ###
+FROM debian:buster-slim as runner
+
+ARG DEBIAN_FRONTEND
+ARG GOPATH
+
+# Set HOME to a writable directory in case something wants to cache things
+ENV HOME=/tmp
+
+# debos runtime dependencies
+# ca-certificates is required to validate HTTPS certificates when getting debootstrap release file
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        libostree-1-1 \
+        apt-transport-https \
+        ca-certificates \
+        debootstrap \
+        pkg-config \
+        systemd-container \
+        binfmt-support \
+        parted \
+        dosfstools \
+        e2fsprogs \
+        btrfs-progs \
+        bmap-tools \
+        # fakemachine runtime dependencies
+        qemu-system-x86 \
+        qemu-user-static \
+        busybox \
+        linux-image-amd64 \
+        systemd && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY --from=builder $GOPATH/bin/debos /usr/local/bin/debos
```

Comment:
I'm a bit in two minds about that because it may make it non-obvious that really it's just launching debos and you should add command line arguments to debos after the container run.. Maybe it's more obvious for people using docker containers more, but i've never seen a docker container being used like that?

Reply:
Discussion with some folks it does seem like it's a common way for putting docker containers together; So yes sounds like a good suggestions. Probably worht doing it before merging so we don't end up with the container behaviour changing on updates